Sources/HipMobileUI/DesignTime/DesignMode.cs

```diff
@@ -0,0 +1,176 @@
+// limitations under the License.
+
+using PaderbornUniversity.SILab.Hip.Mobile.Shared.BusinessLayer.Models;
+using PaderbornUniversity.SILab.Hip.Mobile.Shared.Common;
+using PaderbornUniversity.SILab.Hip.Mobile.Shared.Common.Contracts;
+using PaderbornUniversity.SILab.Hip.Mobile.Shared.DataAccessLayer;
+using PaderbornUniversity.SILab.Hip.Mobile.UI.Appearance;
+using PaderbornUniversity.SILab.Hip.Mobile.UI.AudioPlayer;
+using PaderbornUniversity.SILab.Hip.Mobile.UI.Contracts;
+using PaderbornUniversity.SILab.Hip.Mobile.UI.DesignTime.Data;
+using PaderbornUniversity.SILab.Hip.Mobile.UI.DesignTime.Services;
+using PaderbornUniversity.SILab.Hip.Mobile.UI.Helpers;
+using PaderbornUniversity.SILab.Hip.Mobile.UI.Location;
+using PaderbornUniversity.SILab.Hip.Mobile.UI.Navigation;
+using PaderbornUniversity.SILab.Hip.Mobile.UI.Pages;
+using PaderbornUniversity.SILab.Hip.Mobile.UI.ViewModels.Pages;
+using PaderbornUniversity.SILab.Hip.Mobile.UI.ViewModels.Views;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Xamarin.Forms;
+
+namespace PaderbornUniversity.SILab.Hip.Mobile.UI.DesignTime
+{
+    /// <summary>
+    /// Supports the design time experience around the Xamarin.Forms Previewer.
+    /// </summary>
+    public static class DesignMode
+    {
+        // view models to use during XAML preview
+        private static readonly Dictionary<Type, object> _viewModels = new Dictionary<Type, object>();
+
+        /// <summary>
+        /// Indicates whether the app is executed in design mode (i.e. by Xamarin.Forms Previewer).
+        /// </summary>
+        /// <remarks>
+        /// This is set to true on app start in AppDelegate (iOS) and MainActivity (Android).
+        /// The Xamarin Forms Previewer does not instantiate those classes, so this flag remains true.
+        /// </remarks>
+        public static bool IsEnabled { get; set; } = true;
+
+        /// <summary>
+        /// To be called in <see cref="Xamarin.Forms.Page"/>-constructors.
+        /// Assigns a sample view model to the view's BindingContext, if the app is executed in design mode.
+        /// If (1) no sample view model is available, (2) the view does not implement <see cref="IViewFor{T}"/>,
+        /// or (3) the app is running for real (on a device or in an emulator), this method has no effect.
+        /// </summary>
+        /// <remarks>
+        /// This method should be called in the constructor of each page, right after the
+        /// "InitializeComponent()"-call.
+        /// </remarks>
+        /// <param name="view">View to set up for previewing</param>
+        public static void Initialize<TView>(TView view) where TView : BindableObject, IViewFor
+        {
+            if (view == null)
+                throw new ArgumentNullException(nameof(view));
+
+            var viewForType = view.GetType().GetTypeInfo().ImplementedInterfaces.FirstOrDefault(ii =>
```

Comment:
Should all of this work not also be skipped if `!IsEnabled`? Of course I haven't run any benchmarks, but especially using reflection might have an effect on the app's performance.

Reply:
Yep, I agree